curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,530 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
+
+```javascript
+function count(n) {
+    if (n === 0) return;
+    console.log(n);
+    count(n - 1);
+}
+count(3);
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+1, 2, 3
 
 ---
 
-Placeholder distractor 2
+3, 3, 3
 
 ---
 
-Placeholder distractor 3
+Nothing
 
 #### --answer--
 
-Placeholder answer
+3, 2, 1
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the following function return?
+
+```javascript
+function factorial(n) {
+  if (n === 0) return 1;
+  return n * factorial(n - 1);
+}
+console.log(factorial(3));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+3
 
 ---
 
-Placeholder distractor 2
+0
 
 ---
 
-Placeholder distractor 3
+1
 
 #### --answer--
 
-Placeholder answer
+6
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a tail-recursive function?
 
 #### --distractors--
 
-Placeholder distractor 1
+A function that only calls itself once.
 
 ---
 
-Placeholder distractor 2
+A function with multiple base cases.
 
 ---
 
-Placeholder distractor 3
+A function that uses a while loop.
 
 #### --answer--
 
-Placeholder answer
+A function where the recursive call is the last operation.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this code print?
+
+```javascript
+function sayHello(n) {
+  if (n <= 0) return;
+  console.log("Hello");
+  sayHello(n - 1);
+}
+sayHello(3);
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+1 time "Hello".
 
 ---
 
-Placeholder distractor 2
+An error.
 
 ---
 
-Placeholder distractor 3
+Infinite "Hello".
 
 #### --answer--
 
-Placeholder answer
+3 times "Hello".
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How many times will the function call itself?
+
+```javascript
+function mystery(n) {
+    if (n <= 1) return 1;
+    return mystery(n - 2);
+}
+mystery(5);
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+2
 
 ---
 
-Placeholder distractor 2
+5
 
 ---
 
-Placeholder distractor 3
+4
 
 #### --answer--
 
-Placeholder answer
+3
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What feature do browsers implement to prevent infinite recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+Memory Cleanup.
 
 ---
 
-Placeholder distractor 2
+Automatic loop breaking.
 
 ---
 
-Placeholder distractor 3
+Thread termination.
 
 #### --answer--
 
-Placeholder answer
+Maximum call stack size
+    - Browsers limit the call stack to prevent infinite recursion.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT true about recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+It must have a base case.
 
 ---
 
-Placeholder distractor 2
+It can replace loops.
 
 ---
 
-Placeholder distractor 3
+It can be memory-intensive.
 
 #### --answer--
 
-Placeholder answer
+It always returns 0.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In which language is the recursive function `factorial` usually written like this?
```

Comment:
this quiz is for recursion is JavaScript, bringing in other languages would be confusing